src/worker/ingestion/process-event.ts

```diff
@@ -237,12 +239,30 @@ export class EventsProcessor {
                 `Could not find person with distinct id "${distinctId}" in team "${teamId}", even after trying to insert them`
             )
         }
+
         const updatedProperties: Properties = { ...propertiesOnce, ...personFound.properties, ...properties }
 
-        if (equal(personFound.properties, updatedProperties)) {
+        let incrementedPropertiesQueryResult: QueryResult | null = null
+
+        const areTherePropsToIncrement = !!Object.keys(incrementProperties).length
+
+        if (areTherePropsToIncrement) {
+            incrementedPropertiesQueryResult = await this.db.incrementPersonProperties(personFound, incrementProperties)
+        }
+
+        const arePersonsEqualExcludingIncrement = equal(personFound.properties, updatedProperties)
+
+        // CH still needs to update if there are $increment props but Postgres has already done so
+        if (arePersonsEqualExcludingIncrement && (!this.db.kafkaProducer || !areTherePropsToIncrement)) {
             return personFound
         }
 
+        if (incrementedPropertiesQueryResult && incrementedPropertiesQueryResult.rows.length > 0) {
+            for (const [key, val] of Object.entries(incrementedPropertiesQueryResult.rows[0].properties)) {
+                updatedProperties[key] = val
+            }
+        }
```

Comment:
Could this be simplified to just `Object.assign(updatedProperties, incrementedPropertiesQueryResult.rows[0].properties)`... or then with some `{...updatedProperties, ...other}` logic?

Reply:
ah yes great point! This could definitely be simplified, reused some logic here